workloads/opea/chatqna/README.md

```diff
@@ -0,0 +1,101 @@
+
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volumes.yaml
+
+```
+``` 
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/persistent_volume_claims.yaml
+
+```
+* Check persistent volume claims are bound to the volumes:
+```
+NAME                          STATUS   VOLUME                       CAPACITY   ACCESS MODES   
+chatqna-megaservice-pvc-0   Bound    chatqna-megaservice-pv-0   100Mi      RWO            
+chatqna-megaservice-pvc-1   Bound    chatqna-megaservice-pv-1   100Mi      RWO            
+chatqna-megaservice-pvc-2   Bound    chatqna-megaservice-pv-2   100Mi      RWO            
+```
+2. From Web Console
+
+Navigate to ```Storage``` -> ```PersistentVolumes```. Click on ```CreatePersistentVolume``` and copy the above yaml. Repeat the same for ```PersistentVolumeClaims```.
+
+### ChatQnA backend- langchain and redis vector database
+1. Deploy anyuid_rbac required for redis vector db and chatqna megaservice
+
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/security/anyuid_rbac.yaml
+
+```
+
+2. Deploy the redis vector database deployment and service
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/redis_deployment_service.yaml
+
+```
+
+3. Build the chatqna megaservice image
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_buildconfig.yaml
+
+```
+
+4. Setup the Certificate for accessing the https endpoint. 
+  ```
+  ./setup_certificate.sh 
+  ```
+
+5. Deploy the chatqna-megaservice pod
+```
+$ oc apply -f https://raw.githubusercontent.com/intel/intel-technology-enabling-for-openshift/main/workloads/opea/chatqna/chatqna_megaservice_deployment.yaml
+```
+
+### Text Generation Inference for Intel Gaudi on RHOAI
+1. Access ```rhods-dashboard``` link from routes on your web console and click on Data Science Projects. Now, create a project.
+
+2. Click on the project, and choose ```Data Connection``` section. In the fields, add your access and secret keys from minio follow [prerequisites section](#prerequisites). This downloads the model and prepares for serving.
+
+3. Now click on the Settings and choose ```ServingRuntime```. Copy or import the [tgi_gaudi_servingruntime.yaml](../tgi-gaudi-servingruntime.yaml). The [tgi-gaudi](https://github.com/huggingface/tgi-gaudi) serving runtime is used.
+
+4. Go back to your project and choose ```Models```. Choose the servingruntime and model from the dropdown (from step 3). Choose the number of accelerators (from step 3), size of container and the data connection- minio (created from step 2). Add the appropriate folder in your bucket for the data connection.
```

Comment:
What Models will we use here? 
Will we need to download the Model manually or RHOAI will download the Model for us?

Reply:
Models supported in opea readme. tried with llama2. we use minio setup for model storage (in step 2 data connection). rhoai model server downloads from minio